agent/src/beerocks/slave/son_slave_thread.cpp

```diff
@@ -4496,35 +4499,75 @@ bool slave_thread::handle_channel_selection_request(Socket *sd, ieee1905_1::Cmdu
                 // add comma if not last channel in the list, else close list by add curl brackets
                 ss << (((ch_idx + 1) != channel_list_length) ? "," : "}");
 
-                // mark first supported non-dfs channel as selected channel
-                // TODO: need to check that selected channel does not violate radio restriction
-                if (preference == 0 || wireless_utils::is_dfs_channel(*channel)) {
-                    continue;
-                }
-
-                LOG(INFO) << "selected_operating_class=" << int(operating_class);
-                LOG(INFO) << "selected_channel=" << int(*channel);
-                selected_channel         = *channel;
-                selected_operating_class = operating_class;
-
-                // TODO: check that the data is parsed properly after fixing the following bug:
-                // Since sFlags is defined after dynamic list cPreferenceOperatingClasses it cause data override
-                // on the first channel on the list and sFlags itself.
-                // See: https://github.com/prplfoundation/prplMesh/issues/8
+                beerocks::message::sWifiChannel wifi_channel;
+                wifi_channel.channel = *channel;
+                channels_list.push_back(wifi_channel);
             }
             LOG(DEBUG) << ss.str();
+            auto pref = beerocks::message::sPreference({operating_class, preference, reason_code});
+            channel_preferences.push_back({pref, channels_list});
```

Comment:
As I remember from previous commits, we are trying to get rid of additional operators like operator==.

Reply:
> As I remember from previous commits, we are trying to get rid of additional operators like operator==.

Absolutely not. What we have avoided is to add an `operator <=` which is only needed to be able to use `std::map` instead of `std::unordered_map`.